resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -57,6 +58,12 @@ private[yarn] class ExecutorRunnable(
   var rpc: YarnRPC = YarnRPC.create(conf)
   var nmClient: NMClient = _
 
+  val clusterId: Option[String] = try {
```

Comment:
Can we do `Utils.tryLog(YarnConfiguration.getClusterId(conf)).toOption`

Reply:
Nice suggestion. Sorry I missed this while dealing with other stuff. Will address.

Reply:
I took a look at `Utils.tryLog` and that doesn't seem to fit my intention, since it logs error message which end users could get a wrong sign. If `RM cluster id` is a mandatory config we may need to take a different approach (like fail-fast). If that is optional, I think we should not leave an error log.